### PR TITLE
Fix and test paste_cell_replace with multi-selection

### LINKED ISSUE #792: Paste cell and replace broken
from menu, if I select 3 cells (markdown; respective test: 1,2,3) it pasts 3 empty code cells.

### PR DESCRIPTION
**Pull request by Carreau**
closes #792 

---

Let's see if travis is happy with my test.

### CODE REVIEW

**Comment by ellisonbg** [comment]
Travis is not happy :(

**Comment by Carreau** [comment]
relaunched.

**Comment by Carreau** [comment]
Fixed case where past and replace was clicked with a selection range. 

Past and replace will now replace the full selection, instead of replacing only the head cell. 

Test included. 

I also greatly simplified the logic to past, which now just delete and insert at index, instead of relying on trick of "insert above" that follow selection.

File: notebook/static/notebook/js/notebook.js [hunk 1]
```diff
@@ -1380,20 +1380,23 @@ define(function (require) {
...
-            var old_selected = this.get_selected_index();
-            this.select(this.find_cell_index(first_inserted));
-            this.delete_cell(old_selected);
+
+        if (!(this.clipboard !== null && this.paste_enabled)) {
+            return
+        }
+
+        var selected =  this.get_selected_cells_indices();
+        var insertion_index = selected[0];
```
Comment (jdfreder) [inline notebook/static/notebook/js/notebook.js]: Assumes `get_selected_cells_indices` is least to greatest, is this an okay assumption?  Else `Math.min.apply(Math, selected)`

File: notebook/static/notebook/js/notebook.js [hunk 2]
```diff
@@ -1380,20 +1380,23 @@ define(function (require) {
...
+        this.delete_cells(selected);
+
+        for (var i=this.clipboard.length-1; i >= 0; i--) {
+            debugger;
```
Comment (jdfreder) [inline notebook/static/notebook/js/notebook.js]: Don't forget to remove this!

File: notebook/static/notebook/js/notebook.js [hunk 3]
```diff
@@ -1380,20 +1380,23 @@ define(function (require) {
...
+            var cell_data = this.clipboard[i];
+            var new_cell = this.insert_cell_at_index(cell_data.cell_type, insertion_index);
+            new_cell.fromJSON(cell_data);
         }
+
+        this.select(insertion_index);
```
Comment (jdfreder) [inline notebook/static/notebook/js/notebook.js]: Hmm, does it make more sense to select the last cell?  `Math.max.apply(Math, selected)`

**Comment by jdfreder** [comment]
Couple comments inline

File: notebook/static/notebook/js/notebook.js (hunk 1 above)
Comment (Carreau) [inline notebook/static/notebook/js/notebook.js]: Yeah, I think we can assume that.

File: notebook/static/notebook/js/notebook.js (hunk 2 above)
Comment (Carreau) [inline notebook/static/notebook/js/notebook.js]: oops.

File: notebook/static/notebook/js/notebook.js (hunk 3 above)
Comment (jdfreder) [inline notebook/static/notebook/js/notebook.js]: Using our cursor analogy, doesn't the head usually end up at the end of the contents you paste?

Comment (Carreau) [inline notebook/static/notebook/js/notebook.js]: Yes, fixed (and tested)

**Comment by jdfreder** [comment]
Awesome, :+1: when green

**Comment by jdfreder** [comment]
Gah, had to restart Travis twice because of illegitimate test failures

**Pull request merged**